### PR TITLE
19852: Plan Match v1.1

### PR DESCRIPTION
**Pull request by sarathpyerra**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [19852](https://devops.dchbx.org/redmine/issues/19852) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-1 |
| Development Story Points | n/a |
| Peer Review Story Points | 1.5 |
| Ticket Author | @davidison |

### CODE REVIEW

File: app/helpers/insured/plan_filter_helper.rb [hunk 1]
```diff
@@ -6,4 +6,11 @@ def find_my_doctor
       link_to('Find Your Doctor', 'https://dc.checkbookhealth.org/dcshop/', target: '_blank')
     end
   end
+
+  def estimate_your_costs
+    if @market_kind == "shop" && @coverage_kind == "health"
+      link_to('Estimate Your Costs', @dc_checkbook_url , target: '_blank')
```
Comment (Battula) [inline app/helpers/insured/plan_filter_helper.rb]: 1. Use l10n

File: config/settings.yml [hunk 2]
```diff
@@ -113,12 +113,16 @@ aca:
...
-    base_url:  "https://dc.checkbookhealth.org"
-    url: "https://dc.checkbookhealth.org/shop/dc/"
-
-aptc_errors:
+# Note: Need to replace Remote_access_key with Production keys while deploying to prod.
+    remote_access_key: 'cd535a89-49ea-4ea6-862f-e9558e8bef50' 
+    reference_id: "9F03A78ADF324AFDBFBEF8E838770132"
+    base_url:  "https://staging.checkbookhealth.org"
+    url: "https://staging.checkbookhealth.org"
+    congress_url: "https://dc.checkbookhealth.org/congress/dc/2018/"
```
Comment (Battula) [inline config/settings.yml]: Change the url so that we don't have to update it every year.

Comment (sarathpyerra) [reply config/settings.yml]: This is the static URL provided by checkbook folks.

File: app/models/services/checkbook_services.rb [hunk 3]
```diff
@@ -51,16 +52,23 @@ def employer_contributions
...
         relationship=  relationship_benefit.relationship == "child_under_26" ? "child" : relationship_benefit.relationship
-        premium_benefit_contributions[relationship] = relationship_benefit.premium_pct.to_f
+        premium_benefit_contributions[relationship] = relationship_benefit.premium_pct.to_i
       end
       premium_benefit_contributions
     end
 
+   def carrier_name
+    elected_plans= census_employee.active_benefit_group.plan_year.benefit_groups.flat_map(&:elected_plans) 
+    elected_plans.first.carrier_profile.legal_name.gsub(" ","") if elected_plans
```
Comment (Battula) [inline app/models/services/checkbook_services.rb]: This might result in an incorrect carrier name if employer offers multiple carriers.

Comment (Battula) [inline app/models/services/checkbook_services.rb]: What if employer offers more than one carrier ?

File: app/models/services/checkbook_services.rb [hunk 4]
```diff
@@ -51,16 +52,23 @@ def employer_contributions
...
+   end
+
+
+
     def build_family
-      family = [{'dob': census_employee.dob,'relationship': 'self'}]
+      family = [{'dob': census_employee.dob.strftime("%Y-%d-%m") ,'relationship': 'self'}]
       census_employee.try(:census_dependents).each do |dependent|
```
Comment (Battula) [inline app/models/services/checkbook_services.rb]: Please do not use try

File: app/controllers/insured/plan_shoppings_controller.rb [hunk 5]
```diff
@@ -164,6 +164,14 @@ def show
       session[:elected_aptc] = 0
     end
 
+    if params[:market_kind] == 'shop'
+      is_congress_employee = @person.employee_roles.detect do |employee_role| 
+        employee_role.employer_profile.active_plan_year.benefit_groups.any?{|bg| bg.is_congress}
```
Comment (Battula) [inline app/controllers/insured/plan_shoppings_controller.rb]: This line will fail when plan year is in any other state(published, enrolling, enrolled) other than active.

File: app/models/services/checkbook_services.rb [hunk 6]
```diff
@@ -3,46 +3,47 @@
...
+              :body => construct_body.to_json,
+              :headers => { 'Content-Type' => 'application/json' } )
+      uri = @result.parsed_response["URL"]
+      if uri.present?
+        return uri
+      else
+        raise "Unable to generate url"
+      end
+      rescue Exception => e
+        puts "Exception #{e}"
```
Comment (Battula) [inline app/models/services/checkbook_services.rb]: Use Rails.logger.error

File: spec/models/census_employee_spec.rb [hunk 7]
```diff
@@ -901,7 +901,7 @@
       allow(notice_trigger).to receive_message_chain(:notice_builder,:camelize,:constantize,:new).and_return(builder)
       allow(notice_trigger).to receive_message_chain(:notice_trigger_element_group,:notice_peferences).and_return({})
       allow(ApplicationEventKind).to receive_message_chain(:where,:first).and_return(double("ApplicationEventKind",{:notice_triggers => notice_triggers,:title => "title",:event_name => "OutOfPocketNotice"}))
-      allow_any_instance_of(CheckbookServices::PlanComparision).to receive(:generate_url).and_return("http://temp.url")
+      allow_any_instance_of(Services::CheckbookServices::PlanComparision).to receive(:generate_url).and_return("fake_url")
```
Comment (Kevin-Wei) [inline spec/models/census_employee_spec.rb]: wow no

File: spec/controllers/insured/plan_shoppings_controller_spec.rb [hunk 8]
```diff
@@ -570,8 +574,10 @@
           allow(tax_household).to receive(:total_aptc_available_amount_for_enrollment).and_return(111)
           allow(family).to receive(:enrolled_hbx_enrollments).and_return([])
           allow(person).to receive(:active_employee_roles).and_return []
+          allow(person).to receive(:employee_roles).and_return []
           allow(hbx_enrollment).to receive(:coverage_kind).and_return 'health'
           allow(hbx_enrollment).to receive(:kind).and_return 'shop'
+          allow_any_instance_of(Services::CheckbookServices::PlanComparision).to receive(:generate_url).and_return("http://temp.url")
```
Comment (Kevin-Wei) [inline spec/controllers/insured/plan_shoppings_controller_spec.rb]: this is not a good sign

**Pull request merged**